common4j/src/main/com/microsoft/identity/common/java/authscheme/PopAuthenticationSchemeWithClientKeyInternal.java

```diff
@@ -0,0 +1,144 @@
+import com.nimbusds.jose.util.Base64URL;
+
+import org.json.JSONObject;
+
+import java.net.URL;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import lombok.experimental.Accessors;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Internal representation of PoP Authentication Scheme where Key is owned by
+ * Client application.
+ */
+@SuperBuilder(toBuilder = true)
+@EqualsAndHashCode(callSuper = true)
+@Accessors(prefix = "m")
+public class PopAuthenticationSchemeWithClientKeyInternal
+        extends TokenAuthenticationScheme
+        implements IPoPAuthenticationSchemeParams {
+
+    private static final long serialVersionUID = 788393037295696359L;
+    /**
+     * The name of this auth scheme as supplied in the Authorization header value.
+     */
+    public static final String SCHEME_POP_WITH_CLIENT_KEY = "PoP_With_Client_Key";
+
+    @SerializedName(HTTP_METHOD)
+    private String mHttpMethod;
+
+    @SerializedName(URL)
+    private URL mUrl;
+
+    @SerializedName(NONCE)
+    private String mNonce;
+
+    @SerializedName(CLIENT_CLAIMS)
+    private String mClientClaims;
+
+    @SerializedName(KID)
+    private String mKid;
+
+    /**
+     * Constructor for gson use.
+     */
+    PopAuthenticationSchemeWithClientKeyInternal() {
+        super(SCHEME_POP_WITH_CLIENT_KEY);
+    }
+
+    /**
+     * Constructs a new PopAuthenticationSchemeWithClientKeyInternal.
+     *
+     * @param httpMethod   The HTTP method associated with this request. Optional.
+     * @param url          The resource URL of future-recipient of this SHR.
+     * @param nonce        Client nonce value; for replay protection.
+     * @param clientClaims Optional claims provided by the caller to embed in the client_claims
+     *                     property of the resulting SHR.
+     */
+    public PopAuthenticationSchemeWithClientKeyInternal(@Nullable final String httpMethod,
+                                                        @NonNull final URL url,
+                                                        @Nullable final String nonce,
+                                                        @Nullable final String clientClaims,
+                                                        @NonNull final String kid) {
+        super(SCHEME_POP_WITH_CLIENT_KEY);
+        mHttpMethod = httpMethod;
+        mUrl = url;
+        mNonce = nonce;
+        mClientClaims = clientClaims;
+        mKid = kid;
+    }
+
+    @Override
+    public String getAccessTokenForScheme(@NonNull final String accessToken) {
+        return accessToken;
+    }
+
+    @Override
+    @Nullable
+    public String getHttpMethod() {
+        return mHttpMethod;
+    }
+
+    @Override
+    public URL getUrl() {
+        return mUrl;
+    }
+
+    @Override
+    public String getClientClaims() {
+        return mClientClaims;
+    }
+
+    @Override
+    @Nullable
+    public String getNonce() {
+        return mNonce;
+    }
```

Comment:
Looks like we wrote a lot of code here to write the `getters` even though we have lombok. Why not just add the `@getters` annotation to the class?

Reply:
These were automatically added as part of implementing the interface